src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -1695,8 +1711,8 @@ int wm_vuldet_check_agent_vulnerabilities(agent_software *agents, wm_vuldet_flag
         if (result != 2) {  // Check that this agent has software to analyze
             if (VU_AGENT_REQUEST_LIMIT && i >= VU_AGENT_REQUEST_LIMIT) {
                 // Report a group of agents
-                if (agent_error = wm_vuldet_report_agent_vulnerabilities(db, agents_it, flags, i), agent_error) {
-                    mterror(WM_VULNDETECTOR_LOGTAG, VU_REPORT_ERROR, agent_error, sqlite3_errmsg(db));
+                if (wm_vuldet_report_agent_vulnerabilities(db, agents_it, flags, i) < 0) {
+                    mterror(WM_VULNDETECTOR_LOGTAG, VU_REPORT_ERROR, agents_it->agent_id, sqlite3_errmsg(db));
```

Comment:
When an error happens and `db` is NULL, shouldn't you stop iterating and return?

Reply:
Idem in line 1708: https://github.com/wazuh/wazuh/pull/5015/files#diff-88895483f5107f24a4e15adf36758f39R1708

Reply:
Agree. It should stop iterating because the DB handled has been closed, and the scanner entries in an inconsistency. I suggest you treat this issue in a new PR along with the other places it happens.